plugins/speculation-rules/load.php

```diff
@@ -20,15 +20,67 @@
 	exit;
 }
 
-// Define the constant.
-if ( defined( 'SPECULATION_RULES_VERSION' ) ) {
-	return;
-}
+(
+	/**
+	 * Register this copy of the plugin among other potential copies.
+	 *
+	 * @param string  $global_var_name Global variable name for storing the plugin pending loading.
+	 * @param string  $version         Version.
+	 * @param Closure $load            Callback that loads the plugin.
+	 */
+	static function ( string $global_var_name, string $version, Closure $load ) {
+		if ( ! isset( $GLOBALS[ $global_var_name ] ) ) {
+			$bootstrap = static function () use ( $global_var_name ) {
+				if (
+					isset( $GLOBALS[ $global_var_name ]['load'] )
+					&&
+					$GLOBALS[ $global_var_name ]['load'] instanceof Closure
+				) {
+					call_user_func( $GLOBALS[ $global_var_name ]['load'] );
+					unset( $GLOBALS[ $global_var_name ] );
```

Comment:
Oh, I just realized that this should _not_ be unset. It should rather set a flag to say that the callback has been fired. Otherwise, if another copy of the plugin is loaded later (e.g. in a theme) then it would attempt to load it again, which could cause an error. This doesn't happen in Speculative Loading currently because our load logic includes this check:

```php
if ( defined( 'SPECULATION_RULES_VERSION' ) ) {
	return;
}
```

Reply:
You may also be able to just set `$GLOBALS[ $global_var_name ]['load']` to `null` so it won't fire again.

Reply:
In 7b3ee06278dd0286aa41d774d8d5a0b5b96117f8 I opted to set a `loaded` flag on the array instead. The problem with setting `$GLOBALS[ $global_var_name ]['load']` to `null` is that `isset()` would treat it the same as if `$GLOBALS[ $global_var_name ]['load']` was not set at all.

Reply:
This is obsolete with moving to `after_setup_theme` so this is undone in 27284ab3b05f1872858333898a9a51eaaf376ee3.